apps/etterlatte-grunnlag/src/main/kotlin/grunnlag/GrunnlagService.kt

```diff
@@ -313,37 +333,48 @@ class RealGrunnlagService(
     }
 
     override fun lagreNyePersonopplysninger(
-        sak: Long,
+        sakId: Long,
+        behandlingId: UUID,
         fnr: Folkeregisteridentifikator,
         nyeOpplysninger: List<Grunnlagsopplysning<JsonNode>>,
     ) {
         logger.info("Oppretter et grunnlag for personopplysninger")
-        val gjeldendeGrunnlag = opplysningDao.finnHendelserIGrunnlag(sak).map { it.opplysning.id }
-
-        for (opplysning in nyeOpplysninger) {
-            if (opplysning.id in gjeldendeGrunnlag) {
-                logger.warn(
-                    "Forsøker å lagre personopplysning ${opplysning.id} i sak $sak men den er allerede gjeldende",
-                )
-            } else {
-                opplysningDao.leggOpplysningTilGrunnlag(sak, opplysning, fnr)
-            }
-        }
+        oppdaterGrunnlagOgVersjon(sakId, behandlingId, fnr, nyeOpplysninger)
     }
 
     override fun lagreNyeSaksopplysninger(
         sak: Long,
+        behandlingId: UUID,
         nyeOpplysninger: List<Grunnlagsopplysning<JsonNode>>,
     ) {
         logger.info("Oppretter et grunnlag for saksopplysninger")
+        oppdaterGrunnlagOgVersjon(sak, behandlingId, fnr = null, nyeOpplysninger)
+    }
+
+    private fun oppdaterGrunnlagOgVersjon(
+        sak: Long,
+        behandlingId: UUID,
+        fnr: Folkeregisteridentifikator?,
+        nyeOpplysninger: List<Grunnlagsopplysning<JsonNode>>,
+    ) {
         val gjeldendeGrunnlag = opplysningDao.finnHendelserIGrunnlag(sak).map { it.opplysning.id }
 
-        for (opplysning in nyeOpplysninger) {
-            if (opplysning.id in gjeldendeGrunnlag) {
-                logger.warn("Forsøker å lagre sakopplysning ${opplysning.id} i sak $sak men den er allerede gjeldende")
-            } else {
-                opplysningDao.leggOpplysningTilGrunnlag(sak, opplysning, null)
-            }
+        val hendelsenummer =
+            nyeOpplysninger.mapNotNull { opplysning ->
+                if (opplysning.id in gjeldendeGrunnlag) {
+                    logger.warn("Forsøker å lagre opplysning ${opplysning.id} i sak $sak men den er allerede gjeldende")
+                    null
+                } else {
+                    opplysningDao.leggOpplysningTilGrunnlag(sak, opplysning, fnr)
+                }
+            }.maxOrNull()
+
+        if (hendelsenummer == null) {
+            logger.warn("Hendelsenummer er null – kan ikke oppdatere versjon for behandling (id=$behandlingId)")
+        } else {
+            // TODO: Hva skal vi gjøre dersom det forsøkes å oppdatere versjon som er låst? Bare hoppe over?
```

Comment:
Det er litt vanskelig å si hva som blir best, men jeg ser for meg at vi også kanskje har lyst til å ikke lagre denne opplysningen, siden den ikke blir knyttet til behandlingen og er på en måte en feil på en eller annen måte. Og jeg tenker en 4xx-respons er good her

Reply:
Høres fornuftig ut! Lager POC i neste PR.